components/release.sh

```diff
@@ -39,62 +38,39 @@ images=(
   "ml-pipeline-gcp"
 )
 
-COMMIT_SHA=$1
+TAG_NAME=$1
 FROM_GCR_PREFIX='gcr.io/ml-pipeline-test/'
 TO_GCR_PREFIX='gcr.io/ml-pipeline/'
-REPO=kubeflow/pipelines
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" > /dev/null && pwd)"
 
-if [ -z "$COMMIT_SHA" ]; then
-  echo "Usage: release.sh <commit-SHA>" >&2
+if [ -z "$TAG_NAME" ]; then
+  echo "Usage: release.sh <tag-name>" >&2
   exit 1
 fi
 
-# Checking out the repo
-clone_dir=$(mktemp -d)
-git clone "git@github.com:${REPO}.git" "$clone_dir"
-cd "$clone_dir"
-branch="release-$COMMIT_SHA"
-# Creating the release branch from the specified commit
-release_head=$COMMIT_SHA
-git checkout "$release_head" -b "$branch"
+# KFP repo root
+pushd "$DIR/.."
 
-# Releasing the container images to public. Updating components and samples.
+# Update setup.py VERSION
+sed -i.bak -e "s|VERSION =.\+'|VERSION = '${TAG_NAME}'|g" "components/gcp/container/component_sdk/python/setup.py"
+
+# Updating components and samples.
 for image in "${images[@]}"
 do
   TARGET_IMAGE_BASE=${TO_GCR_PREFIX}${image}
-  TARGET_IMAGE=${TARGET_IMAGE_BASE}:${COMMIT_SHA}
-
-  # Move image from test to prod GCR
-  gcloud container images add-tag --quiet \
-  ${FROM_GCR_PREFIX}${image}:${COMMIT_SHA} ${TARGET_IMAGE}
+  TARGET_IMAGE=${TARGET_IMAGE_BASE}:${TAG_NAME}
 
   # Update the code
-  find components samples -type f | while read file; do sed -i -e "s|${TARGET_IMAGE_BASE}:\([a-zA-Z0-9_.-]\)\+|${TARGET_IMAGE}|g" "$file"; done
+  find components samples -type f | while read file; do
+    sed -i -e "s|${TARGET_IMAGE_BASE}:\([a-zA-Z0-9_.-]\)\+|${TARGET_IMAGE}|g" "$file"
+  done
 done
 
-# Checking-in the container image changes
-git add --all
-git commit --message "Updated component images to version $COMMIT_SHA"
-image_update_commit_sha=$(git rev-parse HEAD)
-
 # Updating the samples to use the updated components
-git diff HEAD~1 HEAD --name-only | while read component_file; do
+git diff --name-only | while read component_file; do
     echo $component_file
     find components samples -type f | while read file; do
-      sed -i -E "s|(https://raw.githubusercontent.com/kubeflow/pipelines/)[^/]+(/$component_file)|\1${image_update_commit_sha}\2|g" "$file";
+      sed -i -E "s|(https://raw.githubusercontent.com/kubeflow/pipelines/)[^/]+(/$component_file)|\1${TAG_NAME}\2|g" "$file";
```

Comment:
Are you sure that this part is correct and the new URLs actually exist? Is the tag created after these changes are committed or before that?

Reply:
I'll try this e2e, if I didn't make a mistake, I think this will work.
For clarification, these urls are updated first. And we tag this commit as that specified version afterwards.

Reply:
The reason I was making two commits before is so that the pipelines have a commit to reference.

Reply:
Maybe given the ad-hoc nature of the component image release, we can exclude this script from the auto-release process.

In the future we can have a better per-component release process (e.g. when you change the code of some component, the image is built and pushed). But for now the ad-hoc component release seems good enough.

Reply:
let's continue the discussion in email thread